sdk/src/azure/core/az_mqtt5_connection.c

```diff
@@ -38,6 +45,29 @@ AZ_NODISCARD az_result az_mqtt5_connection_init(
   _az_PRECONDITION_NOT_NULL(event_callback);
 
   client->_internal.options = options == NULL ? az_mqtt5_connection_options_default() : *options;
+
+  _az_PRECONDITION_RANGE(-1, client->_internal.options.max_connect_attempts, INT16_MAX - 1);
+  _az_PRECONDITION_RANGE(0, client->_internal.options.min_retry_delay_msec, INT32_MAX - 1);
+  _az_PRECONDITION_RANGE(0, client->_internal.options.max_retry_delay_msec, INT32_MAX - 1);
+  _az_PRECONDITION_RANGE(0, client->_internal.options.max_random_jitter_msec, INT32_MAX - 1);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.options.hostname, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.options.client_id_buffer, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.options.username_buffer, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.options.password_buffer, 0, true);
+#ifndef AZ_NO_PRECONDITION_CHECKING
+  for (int i = 0; i < client->_internal.options.client_certificate_count; i++)
+  {
+    _az_PRECONDITION_VALID_SPAN(client->_internal.options.client_certificates[i].cert, 1, false);
+    _az_PRECONDITION_VALID_SPAN(client->_internal.options.client_certificates[i].key, 1, false);
+  }
+#endif // AZ_NO_PRECONDITION_CHECKING
+
+  client->_internal.client_certificate_index = 0;
+  client->_internal.reconnect_counter = 0;
+  client->_internal.disconnect_handshake_timeout_msec
```

Comment:
Is it worth setting this here vs just using the define where needed?

Reply:
Good point, the smaller we can make options the better. I'll probably do something similar in the future for many of the defaults